superset/viz.py

```diff
@@ -106,10 +106,11 @@ def get_df(self, query_obj=None):
                 timestamp_format = dttm_col.python_date_format
 
         # The datasource here can be different backend but the interface is common
-        self.results = self.datasource.query(**query_obj)
+        query_str, engine, start_dttm = self.datasource.get_query_str(**query_obj)
```

Comment:
At first sight it seems like you wouldn't need start_dttm and engine to be returned by `get_query_str`. Functions returning tuples are generally a bad idea, it points in the direction of making a more targeted function or returning an object. In this case the former.